.circleci/config.yml

```diff
@@ -1,26 +1,27 @@
 version: 2
 
 jobs:
-  clone_repo_and_resolve_dependencies:
+  clone_repo_and_compile_project:
     machine: true
     working_directory: ~/repo
     steps:
       - checkout
       - run:
-          name: Clone repo & resolve dependencies
+          name: Clone repo & compile project
```

Comment:
the name seems confusing since nothing is compiled in this step.

Reply:
Will update this. Trying to comparing resolve vs compile. Compile does the caching of classes as well and resolve only caches external libs. From observations, both seem to give effectively similar runtime overall. But compile takes more time in step one and less over the following parallel splits and resolve is other way round. Let's check the run times for both and decide on this.